src/composable/axios.ts

```diff
@@ -0,0 +1,40 @@
+import { ref } from 'vue'
+import axios from 'axios'
+const api = axios.create({
+  baseURL: import.meta.env.VITE_API_URL
+})
+export const useAxios = async (url: string, method: string, payload: any = null) => {
+  const data = ref(null)
+  const error = ref('')
+  try {
+    let response = null
+    switch (method) {
```

Comment:
This whole switch case can be simplified with this

```
response = await api[method](url, payload)
```

But we need to check that method is only a supported value.